examples/async.rs

```diff
@@ -0,0 +1,294 @@
+//! uses the `octocrab` crate to fetch a list of pull requests from the GitHub API. You will need an
+//! environment variable named `GITHUB_TOKEN` with a valid GitHub personal access token. The token
+//! does not need any special permissions.
+//!
+//! This example does not cover message passing between threads, it only demonstrates how to manage
+//! shared state between the main thread and a background task, which acts mostly as a one-shot
+//! fetcher. For more complex scenarios, you may need to use channels or other synchronization
+//! primitives.
+//!
+//! A simple app might have multiple widgets that fetch data from different sources, and each widget
+//! would have its own background task to fetch the data. The main thread would then render the
+//! widgets with the latest data.
+//!
+//! The latest version of this example is available in the [examples] folder in the repository.
+//!
+//! Please note that the examples are designed to be run against the `main` branch of the Github
+//! repository. This means that you may not be able to compile with the latest release version on
+//! crates.io, or the one that you have installed locally.
+//!
+//! See the [examples readme] for more information on finding examples that match the version of the
+//! library you are using.
+//!
+//! [Ratatui]: https://github.com/ratatui-org/ratatui
+//! [examples]: https://github.com/ratatui-org/ratatui/blob/main/examples
+//! [examples readme]: https://github.com/ratatui-org/ratatui/blob/main/examples/README.md
+use std::{
+    sync::{Arc, RwLock},
+    time::Duration,
+};
+
+use color_eyre::Result;
+use futures::StreamExt;
+use octocrab::{
+    params::{pulls::Sort, Direction},
+    OctocrabBuilder, Page,
+};
+use ratatui::{
+    crossterm::event::{Event, EventStream, KeyCode},
+    layout::Offset,
+    prelude::{Buffer, Constraint, Line, Modifier, Rect, Stylize},
+    widgets::{
+        Block, BorderType, HighlightSpacing, Row, StatefulWidget, Table, TableState, Widget,
+    },
+};
+use tokio::time::interval;
+
+use self::terminal::Terminal;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    color_eyre::install()?;
+    init_octocrab()?;
+    let terminal = terminal::init()?;
+    let app_result = App::default().run(terminal).await;
+    terminal::restore();
+    app_result
+}
+
+fn init_octocrab() -> Result<()> {
+    // dotenvy::dotenv()?; // uncomment if you want to use a .env file to load the token
+    let token = std::env::var("GITHUB_TOKEN")?;
+    let crab = OctocrabBuilder::new().personal_token(token).build()?;
+    octocrab::initialise(crab);
+    Ok(())
+}
+
+#[derive(Debug, Default)]
+struct App {
+    should_quit: bool,
+    pulls: PullRequestsWidget,
+}
+
+impl App {
+    const FRAMES_PER_SECOND: f32 = 60.0;
+
+    pub async fn run(mut self, mut terminal: Terminal) -> Result<()> {
+        self.pulls.run();
+
+        let mut interval = interval(Duration::from_secs_f32(1.0 / Self::FRAMES_PER_SECOND));
```

Comment:
```suggestion
        let mut interval = tokio::time::interval(Duration::from_secs_f32(1.0 / Self::FRAMES_PER_SECOND));
```

Reply:
I have a minor preference of inlining this import in examples, it is easier to look for `tokio` related code

Reply:
502508eb6